Library/ViewModels/DashboardViewModel.swift

```diff
@@ -87,34 +102,69 @@ public protocol DashboardViewModelType {
 public final class DashboardViewModel: DashboardViewModelInputs, DashboardViewModelOutputs,
   DashboardViewModelType {
 
-    public init() {
+  //Project selected by user in the drawer or by push for the project message thread
+  private let selectedProjectProperty: SignalProducer<Param?, NoError>
+
+  //Selected project or the first project in the projects collection if no project was selected
+  private let selectProjectPropertyOrFirst = MutableProperty<Param?>(nil)
+
+  //Last recieved push for message or nil if view is about to disappear
+  private let messageThreadReceived = MutableProperty<(Param, MessageThread)?>(nil)
+
+  public init() {
     let projects = self.viewWillAppearAnimatedProperty.signal.filter(isFalse).ignoreValues()
       .switchMap {
         AppEnvironment.current.apiService.fetchProjects(member: true)
           .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
           .demoteErrors()
           .map { $0.projects }
+          .prefix(value: [])
     }
 
+    self.selectedProjectProperty = SignalProducer.merge(
+      self.switchToProjectProperty.producer,
+      self.messageThreadNavigatedProperty.producer.skipNil().map(first)
+    )
+
+    self.selectProjectPropertyOrFirst <~ SignalProducer.combineLatest(
```

Comment:
Gonna try to explore some alternate ways of handling this tomorrow. We have a few isolated cases of using one-off producers for things, but it's definitely the exception and not the rule. Curious if we can find a more direct way of handling signals here.